tools/extras/install_portaudio.sh

```diff
@@ -82,7 +82,7 @@ if [ -z "$MACOS" ]; then
 fi
 
 ./configure --prefix=`pwd`/install --with-pic
-sed -i.bk '40s:src/common/pa_ringbuffer.o::g; 40s:$: src/common/pa_ringbuffer.o:' Makefile
+perl -i.bk -pe 's:src/common/pa_ringbuffer.o:: if $.==40; s:$: src/common/pa_ringbuffer.o: if $.==40' Makefile
```

Comment:
What happens here is a filename removed from a specific Makefile line, which actually is
```
OTHER_OBJS = src/hostapi/oss/pa_unix_oss.o src/os/unix/pa_unix_hostapis.o src/os/unix/pa_unix_util.o src/common/pa_ringbuffer.o
```
To make the script more robust, let's better match it by content (I noted that above regarding an .m file):
```
perl -i -pe 's:src/common/pa_ringbuffer.o:: if /^OTHER_OBJS\s*=/' Makefile
```
The `/^OTHER_OBJS\s*=/` regex applies the substitution only to this assignment line. It is also unnecessary to handle the space specially, as make would not worry if you leave a whitespace at end of line.

Not our case, but if you wanted to remove multiple pattern, the alternative match operator `|` would help, like
```
perl -pe 's:src/common/pa_ringbuffer.o|src/common/something_else.o::g if /^OTHER_OBJS /' Makefile
```
This has a `g` at the end, as the goal is to apply it to all matches in the same string.

Again, this is not our case; just to note a technique.